### ISSUE TITLE
Colormanagement broken in 1.3.0

### ISSUE THREAD

**Issue by MUWalter** [iina/iina#3929]
![Bildschirmfoto 2022-08-28 um 22 52 21](https://user-images.githubusercontent.com/29971407/187094104-52b14337-a578-459e-bda2-c4873f217eba.png)

This is how I usually tell Iina to use the correct ICC-Profile for my monitor. It all worked well and Iina gave me correct colors and gamut until I updated from 1.2.0 to 1.3.0. Colors were off (no color management any more). Does anybody have the same issue?

**Comment by low-batt** [comment]
No need to ask others. I looked at the code. New HDR related code added for 1.3.0 in `VideoView` is not checking to see if `icc-profile` has been specified in `Advanced` preferences.

@CarterLi Thoughts on this one?

I'm thinking the code in `VideoView` needs to be updated to check and see if `icc-profile` has been specified in `Advanced mpv options` and use the user supplied value.

There are additional mpv ICC profile related options such as [icc-profile-auto](https://mpv.io/manual/stable/#options-icc-profile-auto). I think the use of that one needs to be handled as well. Other ICC profile options should be tested to see how they interact with the  `VideoView` code.

**Comment by CarterLi** [comment]
Did you have `Load ICC profile` preference unchecked?

<img width="814" alt="image" src="https://user-images.githubusercontent.com/6134068/187108084-c05f3a5e-6419-4552-b589-48a58ee242ce.png">

> New HDR related code added for 1.3.0 in VideoView is not checking to see if icc-profile has been specified in Advanced preferences.

We never had such code.

https://github.com/iina/iina/pull/3701/files?w=1#diff-0da7dedcb527d7cb82822500e08f5a55ee5fb2a7fb99b2bf039dbad1b6787546L268

User setting `icc-profile` in `Advanced mpv options ` will always be overridden if IINA successfully find some file to use. I doubt that the setting ever truly work.

**Comment by MUWalter** [comment]
If I disable "LOAD ICC profile“, the result is even worse and the picture is oversaturated. In Iina 1.2.0 gamma and gamut are as they are supposed to be.

**Comment by low-batt** [comment]
The mpv option in question is [icc-profile](https://mpv.io/manual/stable/#options-icc-profile):
> **--icc-profile=\<file>**
>  &nbsp; &nbsp; &nbsp; &nbsp;Load an ICC profile and use it to transform video RGB to screen output. Needs LittleCMS 2 support
>  &nbsp; &nbsp; &nbsp; &nbsp;compiled in. This option overrides the `--target-prim`, `--target-trc` and
>  &nbsp; &nbsp; &nbsp; &nbsp;`--icc-profile-auto` options.

This option is not listed in the "Reserved options" section of the IINA wiki page [MPV Options and Properties](https://github.com/iina/iina/wiki/MPV-Options-and-Properties#reserved-options).

Some very quick tests checking the handling of `icc-profile` as reported in `mpv.log`...

### IINA 1.2.0

No mpv options configured:
```text
low-batt@gag 1.2.0-no-config$ grep -i ICC mpv.log
[   0.556][v][cplayer] Set property: icc-profile="/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc" -> 1
[   0.733][d][global] user path: '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc' -> '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
[   0.733][v][libmpv_render] Opening ICC profile '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
```

With `icc-profile` set to `/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc` in `Additional mpv options`:
```text
low-batt@gag 1.2.0-with-config$ grep -i ICC mpv.log
[   0.254][d][global] user path: '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc' -> '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc'
[   0.254][v][libmpv_render] Opening ICC profile '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc'
[   0.492][v][cplayer] Set property: icc-profile="/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc" -> 1
[   0.644][d][global] user path: '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc' -> '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
[   0.644][v][libmpv_render] Opening ICC profile '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
```

The `icc-profile` setting in `Additional mpv options` was passed to mpv, but was then overridden.

### IINA 1.3.0

IINA 1.3.0 adds a new  `Load ICC profile` preference in `Video/Audio`. This preference is enabled by default.

#### Load ICC Profile Enabled

No mpv options configured:
```text
[   0.493][v][cplayer] Set property: icc-profile="/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc" -> 1
[   0.497][d][global] user path: '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc' -> '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
[   0.497][v][libmpv_render] Opening ICC profile '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
```

With `icc-profile` set to `/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc` in `Additional mpv options` :
```text
low-batt@gag 1.3.0-with-config$ grep -i ICC mpv.log
[   0.245][d][global] user path: '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc' -> '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc'
[   0.245][v][libmpv_render] Opening ICC profile '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc'
[   0.422][v][libmpv_render] Detected ICC profile contrast: inf
[   0.503][v][cplayer] Set property: icc-profile="/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc" -> 1
[   0.512][d][global] user path: '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc' -> '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
[   0.512][v][libmpv_render] Opening ICC profile '/Library/ColorSync/Profiles/Displays/Color LCD-37D8832A-2D66-02CA-B9F7-8F30A301B230.icc'
```

The `icc-profile` setting in `Additional mpv options`  was passed to mpv, but was then overridden.

#### Load ICC Profile Disabled

No mpv options configured:
```text
[   0.437][v][cplayer] Set property: icc-profile="" -> 1
```

With `icc-profile` set to `/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc` in `Additional mpv options` :
```text
low-batt@gag 1.3.0-with-config-no-icc$ grep -i ICC mpv.log
[   0.248][d][global] user path: '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc' -> '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc'
[   0.248][v][libmpv_render] Opening ICC profile '/Library/ColorSync/Profiles/Displays/LG UltraFine-E38F6C1F-0BB4-4054-BFA3-691358E5FE0F.icc'
[   0.423][v][cplayer] Set property: icc-profile="" -> 1
```

The `icc-profile` setting in `Additional mpv options`  was passed to mpv, but was then cleared.

### Conclusions

I'm confused. This was a very quick test, so I could have messed up, but it sure looks like setting `icc-profile` in `Additional mpv options` is not working in both 1.2.0 and 1.3.0.

Maybe the change in behavior is due to other settings?

**Comment by low-batt** [comment]
This was fixed in IINA 1.4.0-beta1.
